### PR TITLE
Terms of Service

### PR DESCRIPTION
**Pull request by V13Axel**
Opening this as a PR to comment on things

### CODE REVIEW

File: app/Http/Controllers/AgreementController.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Auth;
+use App\Agreement;
+
+class AgreementController extends Controller
+{
+
+    private function get_agreement(){
+        $tos = Agreement::where("in_effect_at", "<=", now())->latest()->first();
```
Comment (V13Axel) [inline app/Http/Controllers/AgreementController.php:13]: Inclined to think the logic for "how we get latest agreement in effect" should be a method on `Agreement`. Something like `Agreement->current()` that would just get the most current agreement that is in effect (since any previous ones aren't any longer).

File: resources/views/pages/prompt-tos.blade.php [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+		@media screen and (max-width: 768px) {
+			.btn {
+				width: 100%;
+			}
+		}
+
+		.scroll_box{
+			overflow-y: scroll;
+			height:50%;
```
Comment (V13Axel) [inline resources/views/pages/prompt-tos.blade.php:18]: So I see what you're going for here. This should be 50vh, and the `.container` should have its `h-100` removed. In its current state we have an unnecessary scrollbar for the whole page that reveals nothing.

File: resources/views/pages/prompt-tos.blade.php [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+@section("content")
+
+
+	<div class="container p-4 h-100" x-data="agreement_form()">
+
+		<div class="alert alert-info" x-show="!dismissed"><a href="#" class="alert-link" style="float: right;" @click="dismissed = true"><i class="fa fa-times"></i></a> Sorry - a lot has changed in the <a class="alert-link" href="{{ route('whats-new') }}">2.0 update</a>, and you'll need to accept the terms of service before you can continue using the site.</div>
+
+		<h1>{{ $title }}</h1>
+		<p><i>{{ $date }}</i></p>
+		<div class="scroll_box mb-4">{!! Markdown::convertToHtml($markdown); !!}</div>
```
Comment (V13Axel) [inline resources/views/pages/prompt-tos.blade.php:44]: Give it a bit of separation and some room to breathe: `class="scroll_box mb-4 border p-3 rounded"` maybe?

File: resources/views/pages/prompt-tos.blade.php [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+		<p>
+			<input id='acknowledgement' type='checkbox' class='mr-2' @click="agreed = !agreed"> By clicking <strong>Agree and Continue</strong>, I hereby agree and constent to the <a target="_blank" href="{{ route('terms-of-service') }}">Terms of Service</a>, and the <a target="_blank" href="{{ route('privacy-policy') }}">GDPR Privacy Policy</a></li>
```
Comment (V13Axel) [inline resources/views/pages/prompt-tos.blade.php:47]: I think this whole label should be clickable for mobile. Excepting, of course, the links therein.

**Comment by Haxxer** [comment]
**To-do checklist:**
- [ ] Update T&C
- [ ] Update Privacy Policy
- [ ] Marketing email consent on registration
- [ ] Marketing consent checklist on profile page
- [ ] ToC and PP version control, include version number when displaying them
- [ ] Cooling off period for services (still unclear)
- [ ] Online Selling Regulation compliance
    - [ ] Customers must know the available payment methods before starting the order.
    - [ ] Customers must be able to amend details of their order before they submit.
    - [ ] The button by which the customer places an order involving payment must be clearly labelled “pay now” or similar.
    - [ ] The subscription amount and frequency, and how to cancel subscriptions, must be clearly stated at the point of order.
    - [ ] Tell customers in the FAQs how to end the contract by deleting their account.

File: app/Http/Controllers/AgreementController.php [hunk 5]
```diff
@@ -5,34 +5,52 @@
...
+                'version' => $tos->id,
+                'markdown' => Markdown::convertToHtml($tos->content),
+                'intended' => $request->input('intended') ?? 'calendars'
+            ]
+        );
     }
 
-    public function agreement_accepted(Request $request) {
+    public function accept(Request $request) {
+        if(!Arr::has($request->all(), 'policy_acceptance')){
```
Comment (V13Axel) [inline app/Http/Controllers/AgreementController.php:44]: You could just do `if($request->has('policy_acceptance'))`

File: app/Http/Controllers/AgreementController.php [hunk 6]
```diff
@@ -5,34 +5,52 @@
...
+            return redirect('pages.prompt-tos');
+        }
+
         Auth::user()->acceptAgreement();
 
+        if(Arr::has($request->all(), 'marketing_acceptance')){
```
Comment (V13Axel) [inline app/Http/Controllers/AgreementController.php:50]: Same here as above

File: app/Http/Controllers/AgreementController.php [hunk 7]
```diff
@@ -5,34 +5,52 @@
...
+            Auth::user()->setMarketingStatus(true);     // We're not passing false here, as we're not opting out
```
Comment (V13Axel) [inline app/Http/Controllers/AgreementController.php:51]: Maybe worth setting a default value of `true` so that we can just `setMarketingStatus()`?

File: app/Http/Controllers/SettingsController.php [hunk 8]
```diff
@@ -18,9 +19,14 @@ public function profile() {
...
 
+        $subscription = Auth::user()->subscriptions()->active()->first();
+        
+        $renews_at = $subscription ? (new Carbon($subscription->asStripeSubscription()->current_period_end))->toFormattedDateString() : False;
+
         return view('pages.profile', [
             'user' => Auth::user(),
-            'subscription' => Auth::user()->subscriptions()->active()->first(),
+            'subscription' => $subscription,
+            'subscription_renews_at' => $renews_at,
```
Comment (V13Axel) [inline app/Http/Controllers/SettingsController.php:29]: I **think** `->current_period_end` should already be a Carbon object. If it is, I think you should be able to just do

```
$renews_at = $subscription->asStripeSubscription()->current_period_end->toFormattedDateString() ?? False;
```

Give it a try?

File: app/Http/Controllers/SettingsController.php [hunk 9]
```diff
@@ -34,6 +40,8 @@ public function updatePassword(UpdatePasswordRequest $request) {
 
     public function update(StoreUserSettings $request) {
         Auth::user()->setSettings(Arr::only($request->all(), ['dark_theme']));
+        
+        Auth::user()->setMarketingStatus(Arr::has($request->all(), 'marketing_acceptance'));
```
Comment (V13Axel) [inline app/Http/Controllers/SettingsController.php:44]: Can just use `$request->has()`

File: app/Http/Controllers/SettingsController.php [hunk 10]
```diff
@@ -34,6 +40,8 @@ public function updatePassword(UpdatePasswordRequest $request) {
 
     public function update(StoreUserSettings $request) {
         Auth::user()->setSettings(Arr::only($request->all(), ['dark_theme']));
```
Comment (V13Axel) [inline app/Http/Controllers/SettingsController.php:42]: Can update this to just `$request->only(['dark_theme'])`

File: resources/views/pages/faq.blade.php [hunk 11]
```diff
@@ -92,6 +92,19 @@ function faq_page(){
                     {
                         text: "I have forgotten my username / email / password!",
                         answer: `@auth Wait... You're <strong>already logged in, {{ Auth::user()->username }}</strong>! You can just go update your password <a href="{{ route('profile') }}">via your profile</a>. @else We've got your back. Before you try to reset it, remember that you can sign in with your email address <i>or</i> your username. If you've forgotten your password, you can reset it here <a target='_blank' href='{{ route('password.request') }}'>here</a>. If you've forgotten your username or email, get in touch with us at <a target='_blank' href='mailto:contact@fantasy-calendar.com'>contact@fantasy-calendar.com</a> and we'll sort you out! @endauth`
+                    },
+                    {
+                        text: "How do I delete my account?",
+                        answer: `
+                            @auth
+                                <p>We're really sorry to hear that!
```
Comment (V13Axel) [inline resources/views/pages/faq.blade.php:100]: With the change to a question, might want to rephrase this as "We're sad to see you go!"

**Comment by Haxxer** [comment]
To do for @V13Axel:
- [ ] Review T&C and Privacy Policy in detail
- [ ] Look into as to why table in html-converted markdown doesn't work (privacy policy, at the bottom)
